src/Agent.Sdk/Knob/AgentKnobs.cs

```diff
@@ -27,6 +27,13 @@ public class AgentKnobs
             new EnvironmentKnobSource("VSTS_SETUP_ALLOW_MOUNT_TASKS_READONLY"),
             new BuiltInDefaultKnobSource("true"));
 
+        public static readonly Knob SkipPostExeceutionIfTargetContainerStopped = new Knob(
+            nameof(SkipPostExeceutionIfTargetContainerStopped),
+            "If true, skips post-execution step for tasks in case the target container has been stopped",
+            new RuntimeKnobSource("VSTS_SETUP_SKIP_POST_EXECUTION_IF_CONTAINER_STOPPED"),
+            new EnvironmentKnobSource("VSTS_SETUP_SKIP_POST_EXECUTION_IF_CONTAINER_STOPPED"),
+            new BuiltInDefaultKnobSource("true"));
```

Comment:
Let's replace "VSTS_SETUP" here with "AGENT". Also I think the default should be false so this skip behavior is explicitly opt-in. Do you have an example of what failure looks like when the container is stopped and this knob is false?

Reply:
When container is stopped and knob is false task's post will fail error like below:

`Error response from daemon: Container ee9f9e8ef938ed2a0dbb9df02192a1f9d7e4a9349df0fc693395dd9a99d21568 is not running
2021-02-03T16:55:17.4536989Z ##[error]Exit code 1 returned from process: file name 'C:\ProgramData\DockerDesktop\version-bin\docker.EXE', arguments 'exec -i  ee9f9e8ef938ed2a0dbb9df02192a1f9d7e4a9349df0fc693395dd9a99d21568 C:\__a\externals\node\bin\node.exe C:\__w\_temp\containerHandlerInvoker.js'.`

[Reference to pipeline with such error](https://dev.azure.com/v-niezz-testorg/TestProject/_build/results?buildId=1524&view=logs&j=91e27a7b-c0a6-5f10-93f2-53a150985a88&t=a1b46944-0373-4beb-b474-e3e42f38e404)

Reply:
I've changed env variable name and default knob value as you suggested.